register_buildpack.go

```diff
@@ -0,0 +1,149 @@
+package pack
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/url"
+	"os/exec"
+	"runtime"
+	"strings"
+	"text/template"
+
+	"github.com/buildpacks/pack/internal/buildpackage"
+	"github.com/buildpacks/pack/internal/dist"
+	"github.com/buildpacks/pack/internal/registry"
+)
+
+type RegisterBuildpackOptions struct {
+	ImageName string
+	Type      string
+	URL       string
+}
+
+type githubIssue struct {
+	title string
+	body  string
+}
+
+func (c *Client) RegisterBuildpack(ctx context.Context, opts RegisterBuildpackOptions) error {
+	appImage, err := c.imageFetcher.Fetch(ctx, opts.ImageName, false, true)
+	if err != nil {
+		return err
+	}
+
+	label, err := appImage.Label(buildpackage.MetadataLabel)
+	if err != nil {
+		return err
+	}
+
+	c.logger.Debugf("Found image label %s: %s", buildpackage.MetadataLabel, label)
+	var buildpackInfo dist.BuildpackInfo
+	if err = json.Unmarshal([]byte(label), &buildpackInfo); err != nil {
+		return err
+	}
+
+	namespace, name, err := parseID(buildpackInfo.ID)
+	if err != nil {
+		return err
+	}
+
+	id, err := appImage.Identifier()
+	if err != nil {
+		return err
+	}
+
+	buildpack := registry.Buildpack{
+		Namespace: namespace,
+		Name:      name,
+		Version:   buildpackInfo.Version,
+		Address:   id.String(),
+		Yanked:    false,
+	}
+
+	issueURL, err := parseURL(opts.URL)
+	if err != nil {
+		return err
+	}
+
+	issue, err := createGithubIssue(buildpack)
+	if err != nil {
+		return err
+	}
+
+	params := url.Values{}
+	params.Add("title", issue.title)
+	params.Add("body", issue.body)
+	issueURL.RawQuery = params.Encode()
+
+	c.logger.Debugf("Open URL in browser: %s", issueURL)
+	return openBrowser(issueURL.String())
+}
+
+var execCommand = exec.Command
```

Comment:
nit: Move this `var` and `openBrowser()` to bottom of file, so the functions read in order of definition. Feel free to disregard this though